api/server.go

```diff
@@ -158,11 +155,13 @@ func (s *Server) authFunc(ctx context.Context) (context.Context, error) {
 	} else {
 		scope = session.Scope
 	}
-	newCtx = context.WithValue(newCtx, reqKey("scope"), scope)
 
 	if err := s.service.collections.Sessions.Touch(ctx, session); err != nil {
 		return nil, err
 	}
 
+	newCtx := context.WithValue(ctx, reqKey("session"), session)
+	newCtx = context.WithValue(newCtx, reqKey("user"), user)
+	newCtx = context.WithValue(newCtx, reqKey("scope"), scope)
```

Comment:
Want to get a brain dump from you about how `Context` is used by gRPC (or other servers) to manage session data like this. Looks powerful but I don't totally get it.

Reply:
Cool. I'm mostly just following examples, but there's an `PerRPCCredentials` interface that has this method:
```
// GetRequestMetadata gets the current request metadata, refreshing
// tokens if required. This should be called by the transport layer on
// each request, and the data should be populated in headers or other
// context. If a status code is returned, it will be used as the status
// for the RPC. uri is the URI of the entry point for the request.
// When supported by the underlying implementation, ctx can be used for
// timeout and cancellation. Additionally, RequestInfo data will be
// available via ctx to this call.
// TODO(zhaoq): Define the set of the qualified keys instead of leaving
// it as an arbitrary string.
GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error)
```
This essentially maps context values (k/v pairs on any Go context) to headers.